client/src/cb-admin/pages/Settings.js

```diff
@@ -112,15 +123,69 @@ export default class SettingsPage extends React.Component {
           this.props.history.push('/admin/login');
         }
       });
+
+    Visitors.get(this.props.auth)
+      .then((res) => {
+        this.props.updateAdminToken(res.headers.authorization);
+        this.setState({ users: res.data.result });
+        this.state.users.unshift({
```

Comment:
_NEVER_ alter the state directly! Also, this doesn't make sense because `setState` is in general asynchronous. You cannot rely on `users` to have the values in `res.data.result` by the time you get to line 131.